src/savegame_json.cpp

```diff
@@ -1447,6 +1447,15 @@ void vehicle::deserialize(JsonIn &jsin)
 
     data.read("parts", parts);
 
+    // we persist the pivot anchor so that if the rules for finding
+    // the pivot change, existing vehicles do not shift around.
+    // Loading vehicles that predate the pivot logic is a special
+    // case of this, they will load with an anchor of (0,0) which
+    // is what they're expecting.
+    data.read("pivot", pivot_anchor[0]);
+    pivot_anchor[1] = pivot_anchor[0];
+    pivot_rotation[1] = pivot_rotation[0] = fdir;
```

Comment:
No, the vehicle shouldn't shift. The invariant is that the tile under the vehicle's posx/posy (in world coords) is the tile at pivot_anchor\[0\] (in mount coords). So long as that is true initially, any shift in pivot point is handled during vehicle movement so that the vehicle doesn't artificially move due to the change in pivot. The savegame bit is just about making sure that we preserve the previous anchor point across saves even if the method of calculating it changes.